src/Dumper.jl

```diff
@@ -0,0 +1,117 @@
+###-----------------------------------------------------------------------------
+### Copyright (C) 2022- Cursor Insight
+###
+### All rights reserved.
+###-----------------------------------------------------------------------------
+
+module Dumper
+
+###=============================================================================
+### Imports
+###=============================================================================
+
+include("Utilities.jl")
+
+using Base: @kwdef
+using MacroTools: @capture
+using .Utilities: dump_directory
+
+###=============================================================================
+### Implementation
+###=============================================================================
+
+###-----------------------------------------------------------------------------
+### State
+###-----------------------------------------------------------------------------
+
+# TODO maybe use some timestamped folder by default
+@kwdef mutable struct State
+    enabled::Bool = false
+    isabsolute::Bool = false
+    mime::MIME = MIME"text/plain"()
+    directory::String = get(ENV, "DUMP", dump_directory("."))
+    mode::String = "w"
+end
+
+const STATE = State()
+
+###-----------------------------------------------------------------------------
+### Management API
+###-----------------------------------------------------------------------------
+
+function enable!()::Nothing
+    STATE.enabled = true
+    return nothing
+end
+
+function enable!(directory::AbstractString)::Nothing
+    STATE.directory = directory
+    mkpath(STATE.directory)
+    enable!()
+    return nothing
+end
+
+function disable!()::Nothing
+    STATE.enabled = false
+    return nothing
+end
+
+###-----------------------------------------------------------------------------
+### Dump
+###-----------------------------------------------------------------------------
+
+macro dump(variable::Symbol, arguments...)
+    arguments = map(arguments) do argument
+        @capture(argument, key_ = value_) || error("Invalid argument")
+        return key => value
+    end |> Dict
+
+    isabsolute = get(arguments, :isabsolute, :($STATE.isabsolute))
+    path = get(arguments, :path, string(variable))
+    directory = get(arguments, :directory, :($STATE.directory))
+    mime = get(arguments, :mime, :($STATE.mime))
+    mode = get(arguments, :mode, :($STATE.mode))
+
+    return quote
+        if $STATE.enabled
+            let _path = $isabsolute ? $path : joinpath($directory, $path)
+                $save(_path, $mime, $variable; $mode)
+            end
+        end
+    end |> esc
```

Comment:
It is generally not recommended to escape entire expressions, escape individual interpolations instead.

The added benefit, for example, is that you don't have to interpolate `save`.